zerver/lib/export.py

```diff
@@ -1086,7 +1086,7 @@ def fetch_usermessages(
     # for the messages you exported for the users in your realm.
     user_message_query = UserMessage.objects.filter(
         user_profile__realm=realm, message_id__in=message_ids
-    )
+    ).order_by("id")
```

Comment:
Structurally, we want `order_by("message_id", "user_profile_id")` (in that for each user, we always want to sort messages by ID, and it seems fine to have the UserMessage IDs be all tied).  It might be worth doing that ordering, in that we sometimes have situations that insert UserMessage rows after the fact (E.g. when someone stars an old message in a stream they just joined).

Reply:
You can just drop this commit if you want, since the order_by here eventually gets overwritten when we serialize the final data.